### ISSUE TITLE
no_effect lint should also lint calls to const_fn where the result is unused

### ISSUE THREAD

**Issue by oli-obk** [rust-lang-nursery/rust-clippy#2722]
```rust
Vec::new();
```

is an example (soon at least)

**Comment by clarcharr** [comment]
This is rust-lang/rust#50805

**Comment by est31** [comment]
I too think it should be done in rustc. But in order to be in rustc, it may not have false positives. I closed the PR because I couldn't figure out a watertight check.

Thinking about it a bit more we could maybe only lint in non-generic code, aka where all the generic input types are known.
